### PR TITLE
[v18] Refactor/quantity rebased

### PR DESCRIPTION
**Pull request by manoncarbonnel**
Refactored Quantity component according to the new guidelines

### CODE REVIEW

**Review by dpellier — APPROVED** [review]

File: packages/components/src/quantity/src/components/osds-quantity/osds-quantity.spec.ts [hunk 1]
```diff
@@ -19,15 +19,27 @@ describe('spec:osds-quantity', () => {
...
   let controller: OdsQuantityController;
 
+  const ariaLabel: string = 'Quantity WAI-ARIA label';
+  const errorStatus: boolean = true;
+  const name: string = 'quantity-component';
+
   /** base html template (avoid boilerplate) */
-  const baseHtml = (slots: { minus?: string, unnamed?: string, plus?: string }) =>
-    `
+  // const baseHtml = (slots: { minus?: string, unnamed?: string, plus?: string }): string =>
```
Comment (aesteves60) [inline packages/components/src/quantity/src/components/osds-quantity/osds-quantity.spec.ts:27]: Remove comment

File: packages/components/src/quantity/src/components/osds-quantity/osds-quantity.tsx [hunk 2]
```diff
@@ -72,19 +85,19 @@ export class OsdsQuantity implements OdsQuantityAttribute {
     this.onDestroy();
   }
 
-  render() {
+  render(): FunctionalComponent {
     return (
-      <Host>
+      <Host ariaLabel={this.ariaLabel}>
```
Comment (aesteves60) [inline packages/components/src/quantity/src/components/osds-quantity/osds-quantity.tsx:90]: AriaLabel not need with the reflect on the props

File: packages/components/src/quantity/src/components/osds-quantity/osds-quantity.tsx [hunk 3]
```diff
@@ -32,15 +35,25 @@ export class OsdsQuantity implements OdsQuantityAttribute {
 
   @Element() el!: HTMLElement;
 
-  @Prop({ reflect: true }) public disabled?: boolean = DEFAULT_ATTRIBUTE.disabled;
+  @Prop({ reflect: true }) ariaLabel: HTMLElement['ariaLabel'] = DEFAULT_ATTRIBUTE.ariaLabel;
```
Comment (aesteves60) [inline packages/components/src/quantity/src/components/osds-quantity/osds-quantity.tsx:38]: I think you can add props ariaLabelledby

File: packages/components/src/quantity/src/components/osds-quantity/interfaces/attributes.ts [hunk 4]
```diff
@@ -1,10 +1,3 @@
...
-   * Quantity is disabled or not
-   */
-  disabled?: boolean
-}
+import type { OdsCommonFieldAttribute } from '@ovhcloud/ods-common-core';
 
-export {
-  OdsQuantityAttribute,
-};
+export type { OdsCommonFieldAttribute as OdsQuantityAttribute };
```
Comment (aesteves60) [inline packages/components/src/quantity/src/components/osds-quantity/interfaces/attributes.ts:3]: You need to omit some key that is not used in the quantity
`clearable
forbiddenValues
label
loading
placeholder
prefixValue
readonly
required
step
tabindex`

**Review by aesteves60 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**